netidx-protocols/src/rpc.rs

```diff
@@ -503,6 +503,12 @@ pub mod client {
                     )
                     .await
                     .map_err(|_| anyhow!("timeout subscribing to procedure"))??;
+                    if self.0.args.get().is_some() {
```

Comment:
This won't actually fix the race, both threads could pass this point and one will still lose the race. Instead I suggest in this case ignoring the error produced by set, it's entirely redundant anyway, and it's the only way to guarantee that no race happens.